### PR TITLE
Refactor category buttons

### PR DESCRIPTION
**Pull request by ParhamInBinary**
## Problem
#78 

## Solution / Implementation
Refactored multiple buttons to a custom dropdown and implemented correlating tests.

## Testing
test for dropdown functionality
test for redirection when clicking on thumbnail

### CODE REVIEW

File: src/tests/Categories.test.tsx [hunk 1]
```diff
@@ -1,35 +1,56 @@
...
+    // expect and click dropdown
+    const dropdown = screen.getByText('All');
+    expect(dropdown).toBeInTheDocument();
+    await user.click(dropdown);
+
+    // expect correct icon to be displayed
+    expect(screen.getByTestId('dropdownIconUp')).toBeInTheDocument();
+    expect(screen.queryByTestId('dropdownIconDown')).not.toBeInTheDocument();
+
+    // expect Drama category and select it
```
Comment (lysmac) [inline src/tests/Categories.test.tsx:47]: Nitpicking here but surely it should be Adventure rather than Drama? 😅

**Review by lysmac — APPROVED** [review]
Good job with the new category menu!

File: src/tests/Thumbnail.test.tsx [hunk 2]
```diff
@@ -35,4 +39,29 @@ describe('Thumbnail', () => {
...
+
+    // Find the thumbnail of the movie
+    const thumbnails = screen.getAllByRole('movie');
+    expect(thumbnails[0]).toBeInTheDocument();
+
+    // Making sure it found the expected movie
+    expect(thumbnails[0]).toHaveTextContent('The Shawshank Redemption');
+    await user.click(thumbnails[0]);
+
+    const actorHeading = await screen.findByText('The Shawshank Redemption');
```
Comment (gabriel-lugo) [inline src/tests/Thumbnail.test.tsx:64]: Maybe you can consider using a different variable name, such as 'titleHeading', to enhance clarity? :)

File: src/tests/Categories.test.tsx (hunk 1 above)
Comment (ParhamInBinary) [reply src/tests/Categories.test.tsx:47]: Good catch

File: src/tests/Thumbnail.test.tsx (hunk 2 above)
Comment (ParhamInBinary) [reply src/tests/Thumbnail.test.tsx:64]: Will do!

**Pull request merged**